x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/home/index.tsx

```diff
@@ -276,10 +278,28 @@ const AvailablePackages: React.FC = memo(() => {
     [allCategoryPackagesRes]
   );
 
-  const { loading: isLoadingAddableCustomIntegrations, value: addableCustomIntegrations } =
-    useGetAddableCustomIntegrations();
-  const filteredAddableIntegrations = addableCustomIntegrations
-    ? addableCustomIntegrations.filter((integration: CustomIntegration) => {
+  const { loading: isLoadingReplacementCustomIntegrations, value: replacementCustomIntegrations } =
+    useGetReplacementCustomIntegrations();
+
+  const mergedEprPackages: Array<PackageListItem | CustomIntegration> = useMemo(() => {
+    return !isLoadingReplacementCustomIntegrations
+      ? mergeEprPackagesWithReplacements(
+          eprPackages || [],
+          replacementCustomIntegrations || [],
+          selectedCategory as Category
+        )
+      : [];
+  }, [
+    eprPackages,
+    replacementCustomIntegrations,
+    isLoadingReplacementCustomIntegrations,
+    selectedCategory,
+  ]);
```

Comment:
I think it makes sense to extract this into an independent hook that can be unit tested in isolation.